sui/deploy-contract.js

```diff
@@ -23,49 +54,179 @@ async function processCommand(contractName, config, chain, options) {
+    printInfo('Publish transaction digest: ', publishTxn.digest);
 
     const contractConfig = chain.contracts[contractName];
     contractConfig.address = packageId;
-    contractConfig.objects = {
-        [contractName]: contractObject.objectId,
-    };
+    contractConfig.objects = {};
 
     switch (contractName) {
-        case 'GasService':
-            contractConfig.objects.GasCollectorCap = gasCollectorCapObject.objectId;
+        case 'gas_service': {
+            const [GasService, GasCollectorCap] = getObjectIdsByObjectTypes(publishTxn, [
+                `${packageId}::gas_service::GasService`,
+                `${packageId}::gas_service::GasCollectorCap`,
+            ]);
+            contractConfig.objects = { GasService, GasCollectorCap };
             break;
-        default:
+        }
+
+        case 'axelar_gateway': {
+            const { minimumRotationDelay, domainSeparator, policy, previousSigners } = options;
+            const operator = options.operator || keypair.toSuiAddress();
+            const signers = await getSigners(keypair, config, chain, options);
+
+            validateParameters({ isNonEmptyString: { previousSigners, minimumRotationDelay }, isKeccak256Hash: { domainSeparator } });
+
+            const [creatorCap, relayerDiscovery, upgradeCap] = getObjectIdsByObjectTypes(publishTxn, [
+                `${packageId}::gateway::CreatorCap`,
+                `${packageId}::discovery::RelayerDiscovery`,
+                '0x2::package::UpgradeCap',
+            ]);
+
+            const encodedSigners = signersStruct
+                .serialize({
+                    ...signers,
+                    nonce: bytes32Struct.serialize(signers.nonce).toBytes(),
+                })
+                .toBytes();
+
+            const tx = new Transaction();
+
+            const separator = tx.moveCall({
+                target: `${packageId}::bytes32::new`,
+                arguments: [tx.pure(arrayify(domainSeparator))],
+            });
+
+            tx.moveCall({
+                target: `${packageId}::gateway::setup`,
+                arguments: [
+                    tx.object(creatorCap),
+                    tx.pure.address(operator),
+                    separator,
+                    tx.pure.u64(minimumRotationDelay),
+                    tx.pure.u64(options.previousSigners),
+                    tx.pure(bcs.vector(bcs.u8()).serialize(encodedSigners).toBytes()),
+                    tx.object('0x6'),
+                ],
+            });
+
+            if (policy !== '0') {
+                const upgradeType = policy === '128' ? 'only_additive_upgrades' : 'only_dep_upgrades';
+
+                tx.moveCall({
+                    target: `0x2::package::${upgradeType}`,
+                    arguments: [tx.object(upgradeCap)],
+                });
+            }
+
+            const result = await broadcast(client, keypair, tx);
+
+            printInfo('Setup transaction digest: ', result.digest);
```

Comment:
```suggestion
            printInfo('Setup transaction digest', result.digest);
```
printInfo handles this already